homeassistant/components/smartthings/cover.py

```diff
@@ -0,0 +1,146 @@
+from .const import DATA_BROKERS, DOMAIN
+
+DEPENDENCIES = ['smartthings']
+
+VALUE_TO_STATE = {
+    'closed': STATE_CLOSED,
+    'closing': STATE_CLOSING,
+    'open': STATE_OPEN,
+    'opening': STATE_OPENING,
+    'partially open': STATE_OPEN,
+    'unknown': None
+}
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add covers for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [SmartThingsCover(device) for device in broker.devices.values()
+         if broker.any_assigned(device.device_id, COVER_DOMAIN)], True)
+
+
+def get_capabilities(capabilities: Sequence[str]) -> Optional[Sequence[str]]:
+    """Return all capabilities supported if minimum required are present."""
+    from pysmartthings import Capability
+
+    min_required = [
+        Capability.door_control,
+        Capability.garage_door_control,
+        Capability.window_shade
+    ]
+    supported = min_required.copy()
+    supported.extend([
+        Capability.battery,
+        Capability.switch_level
+    ])
+    # Must have one of the min_required
+    if any(capability in capabilities
+           for capability in min_required):
+        return supported
+    return None
+
+
+class SmartThingsCover(SmartThingsEntity, CoverDevice):
+    """Define a SmartThings cover."""
+
+    def __init__(self, device):
+        """Initialize the cover class."""
+        from pysmartthings import Capability
+
+        super().__init__(device)
+        self._device_class = None
+        self._state = None
+        self._state_attrs = None
+        self._supported_features = SUPPORT_OPEN | SUPPORT_CLOSE
+        if Capability.switch_level in device.capabilities:
+            self._supported_features |= SUPPORT_SET_POSITION
+
+    async def async_close_cover(self, **kwargs):
+        """Close cover."""
+        # Same command for all 3 supported capabilities
+        await self._device.close(set_status=True)
+        # State is set optimistically in the commands above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state(True)
+
+    async def async_open_cover(self, **kwargs):
+        """Open the cover."""
+        # Same for all capability types
+        await self._device.open(set_status=True)
+        # State is set optimistically in the commands above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state(True)
+
+    async def async_set_cover_position(self, **kwargs):
+        """Move the cover to a specific position."""
+        position = kwargs.get(ATTR_POSITION)
+        if not self._supported_features & SUPPORT_SET_POSITION \
+                or not position:
```

Comment:
Position argument will always be present. But it can be 0. I think we should remove this part of the check.

Reply:
Zero is a valid value, so good catch.  Simplified this with the other change.